src/Storages/MergeTree/MergeList.cpp

```diff
@@ -65,16 +48,14 @@ MemoryTrackerThreadSwitcher::~MemoryTrackerThreadSwitcher()
 MergeListElement::MergeListElement(
     const StorageID & table_id_,
     FutureMergedMutatedPartPtr future_part,
-    UInt64 memory_profiler_step,
-    UInt64 memory_profiler_sample_probability,
-    UInt64 max_untracked_memory_)
+    const Settings & settings)
```

Comment:
I am a little bit against adding all settings.
It would be probably better to add a structure like `MemoryTrackerSettings` with only relevant one.

Reply:
I thought about this too, but then decided that it is OK for this case, since it is MergeTree code anyway and it does accept `Settings` in lots of other places.

But I can just revert this change, and pass 3 settings separately.
Separate structure is good, but it will require to construct it in 4 places, and this will be more code that this simple thing required.

I'm ok with any solution, `Settings`/`MemoryTrackerSettings` or separate arguments, which one would your prefer?